diff -up sblim-sfcc-2.2.1/backend/cimxml/value.c.orig sblim-sfcc-2.2.1/backend/cimxml/value.c
--- sblim-sfcc-2.2.1/backend/cimxml/value.c.orig	2010-03-15 15:06:51.101378348 +0100
+++ sblim-sfcc-2.2.1/backend/cimxml/value.c	2010-03-15 15:07:33.107450586 +0100
@@ -178,11 +178,13 @@ static char *value2CharsUri(CMPIType typ
       case CMPI_filter:
          break;
 
+      case CMPI_chars:
+         return strdup(value->chars ? (char*)value->chars : "NULL");
+
       case CMPI_string:
       case CMPI_numericString:
       case CMPI_booleanString:
       case CMPI_dateTimeString:
-      case CMPI_classNameString:
          return strdup(value->string ? (char*)value->string->hdl : "NULL");
 
       case CMPI_dateTime:
